### ISSUE TITLE
Functionality doesn't restored after transfer tasker project to another device

### ISSUE THREAD

**Issue by pkjq** [dc297/mqttclpro#65]
When i export project from tasker and then import it at another device.

- In MqttClient i already have subscription to required topic, but subscribscription from tasker doesn't work. It's start to work if open (don't change anything) configuration of profile.

- Publishing doen't work too. And starting to work after open task configuration, wich try to send mqtt message. In this case 'MQTT Publisher Plugin' works correctly.

And if i create App by tasker factory it's doesn't work in another device with symptoms is described above.

**Comment by deepessh** [comment]
Hey

Tasker uses the broker id generated in the app to save tasks/events.

This id might vary from one device to other which might be the reason behind the project not working out of the box.

**Comment by pkjq** [comment]
Can you fix this? 'MQTT Publisher Plugin' works fine. May be ID must be dynamically generated?
Because it's not portable solution :-|.

**Comment by deepessh** [comment]
Let me take a look.

As a workaround, you may edit the xml file exported from tasker and change the broker ID to the correct value.

Cheers!

**Comment by pkjq** [comment]
Yes! Thanks. Workaround is working fine! And i realized what ID you meant.
May be use link with name of connection, not a ID?

**Comment by deepessh** [comment]
Hey, the idea is that you could have multiple connections with the same name, that's why using id is better.

**Comment by pkjq** [comment]
What is it for? We can't control ID, because It's hidden part of implementation details. And as a consequence it's not portable solution without modification of tasker configuration.

If many connections have the same name, may be user wants to connect with backup servers(reserved)?

**Comment by deepessh** [comment]
Hey, there could be multiple use cases when the user wants to have the same name for different connections. The idea was to not restrict the user in any way.

That is how it is by design. Let me think if it could be made portable with minimal changes in the current design.

The id is not hidden per se. It is visible in the brokers list. But I agree, it is not a portable solution.

I would love to hear any suggestions you may have.

Cheers!
Deepesh
